bundles/edu.kit.ipd.sdq.activextendannotations/src/edu/kit/ipd/sdq/activextendannotations/Utility.xtend

```diff
@@ -0,0 +1,71 @@
+package edu.kit.ipd.sdq.activextendannotations
+
+import java.lang.annotation.Target
+import org.eclipse.xtend.lib.macro.Active
+import org.eclipse.xtend.lib.macro.TransformationParticipant
+import java.util.List
+import org.eclipse.xtend.lib.macro.TransformationContext
+import org.eclipse.xtend.lib.macro.declaration.MutableTypeDeclaration
+import org.eclipse.xtend.lib.macro.ValidationParticipant
+import org.eclipse.xtend.lib.macro.ValidationContext
+import org.eclipse.xtend.lib.macro.declaration.MutableClassDeclaration
+import org.eclipse.xtend.lib.macro.declaration.Visibility
+
+/**
+ * Marks a class as a utility class. This makes the class `final`, adds a
+ * private, parameterless constructor and does not allow any other
+ * constructors. Additionally adds errors to instance methods, instance fields
+ * and non-final static fields as these do not belong in a utility class and
+ * were most likely added by accident.
+ * 
+ * Despite these changes to the generated source, this annotation is also meant
+ * as a more declarative way of defining utility classes.
+ */
+@Target(TYPE)
+@Active(UtilityProcessor)
+annotation Utility {
+}
+
+class UtilityProcessor implements TransformationParticipant<MutableTypeDeclaration>, ValidationParticipant<MutableTypeDeclaration> {
+
+	override doTransform(List<? extends MutableTypeDeclaration> annotatedTargetElements,
+		extension TransformationContext context) {
+		for (annotatedType : annotatedTargetElements) {
+			if (annotatedType instanceof MutableClassDeclaration) {
+				annotatedType.final = true
+				val privateParameterlessConstructors = annotatedType.declaredConstructors.filter [
+					visibility == Visibility.PRIVATE && parameters.length == 0
+				]
+				if (privateParameterlessConstructors.length == 0) {
+					annotatedType.addConstructor [
+						visibility = Visibility.PRIVATE
+						body = ''''''
+					]
+				}
+			}
+		}
+	}
+
+	override doValidate(List<? extends MutableTypeDeclaration> annotatedTargetElements,
+		extension ValidationContext context) {
+		for (annotatedType : annotatedTargetElements) {
+			if (annotatedType instanceof MutableClassDeclaration) {
+				annotatedType.declaredConstructors.filter[visibility != Visibility.PRIVATE || parameters.length > 0].
+					forEach [
+						addError("A Utility class may not have a constructor.")
```

Comment:
Because that sounds like you have to add a parameterless constructor to solve the issue, This would no be *wrong*, but unnecessary because `@Utility` creates it for you. If you take this message literally and remove all constructors, everything will be alright.

Although I agree that the message is *technically* wrong.